### PR TITLE
[closed]

### PR DESCRIPTION
**Pull request by CYarros10**
Creating hooks, operators, and test for two Google Cloud generative model APIs.  Namely, the [language model for text generation (PaLM 2)](https://cloud.google.com/vertex-ai/docs/generative-ai/text/test-text-prompts) and the [multimodal model for content generation (Gemini)](https://cloud.google.com/vertex-ai/docs/generative-ai/multimodal/send-chat-prompts-gemini).  This will help data engineers begin to incorporate generative AI into their pipelines. 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/vertex_ai/generative_model.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        self,
+        prompt: str,
+        pretrained_model: str,
+        temperature: float,
+        max_output_tokens: int,
+        top_p: float,
+        top_k: int,
+    ) -> str:
+        """
+        Uses the Vertex AI PaLM API to generate natural language text.
```
Comment (rawwar) [inline airflow/providers/google/cloud/hooks/vertex_ai/generative_model.py]:
```suggestion
        Use the Vertex AI PaLM API to generate natural language text.
```

handles D401 lint issue

File: airflow/providers/google/cloud/hooks/vertex_ai/generative_model.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                "The `delegate_to` parameter has been deprecated before and finally removed in this version"
+                " of Google Provider. You MUST convert it to `impersonate_chain`"
+            )
+        super().__init__(gcp_conn_id=gcp_conn_id, impersonation_chain=impersonation_chain, **kwargs)
+
+    def chat_prompt_request(
+        self, prompt: str, chat: ChatSession | None = None, pretrained_model: str = "gemini-pro"
+    ) -> str:
+        """
+        Uses the Vertex AI Gemini Pro foundation model to generate natural language text.
```
Comment (rawwar) [inline airflow/providers/google/cloud/hooks/vertex_ai/generative_model.py]:
```suggestion
        Use the Vertex AI Gemini Pro foundation model to generate natural language text.
```

File: airflow/providers/google/cloud/operators/vertex_ai/generative_model.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+        self.log.info("Model response: %s", response)
+        self.xcom_push(context, key="prompt_response", value=response)
+
+        return response
+
+
+class MultimodalModelChatOperator(GoogleCloudBaseOperator):
+    """
+    Uses a Vertex AI Multimodal model to generate natural language text.
```
Comment (rawwar) [inline airflow/providers/google/cloud/operators/vertex_ai/generative_model.py]:
```suggestion
    Use a Vertex AI Multimodal model to generate natural language text.
```

**Comment by CYarros10** [comment]
TODO:
- add the integration in airflow/providers/google/provider.yaml
- Add documentation in docs/apache-airflow-providers-google/operators/cloud/
- Basic unit tests for hook and operator

**Comment by CYarros10** [comment]
Build CI Images fail with `ERROR: Could not install packages due to an OSError: [Errno 28] No space left on device` during the airflow installation step

**Pull request closed without merging**